javascript/JS-Async-Functions/js-async-functions_slot-machine/index.js

```diff
@@ -22,6 +22,29 @@ root.append(machine, spinButton, result);
 //                                      ↙️
 spinButton.addEventListener("click", async () => {
   spinButton.disabled = true;
+  try {
+    result.setSpinning();
+    const arrayOfShapes = await Promise.all([
+      wheel1.spin(),
+      wheel2.spin(),
+      wheel3.spin(),
+    ]);
+    console.log(arrayOfShapes);
+    const maxCount = getMaxCount(arrayOfShapes);
+    console.log(maxCount);
+    let points = 0;
+    if (maxCount === 3) {
+      points = 100;
+    } else if (maxCount === 2) {
+      points = 10;
+    } else {
+      points = 0;
+    }
+    console.log(points);
+    result.setResult(points);
+  } catch (error) {
+    result.setMachineChoked();
+  }
```

Comment:
Here you missed the "finally" it must be like :
    result.setResult(points);
  } catch (error) {
    result.setMachineChoked();
  } finally {
    // Enable the spin button after the wheels have stopped or if an error occurred
    spinButton.disabled = false;
  }
});